Zend/zend_API.c

```diff
@@ -4789,12 +4789,28 @@ ZEND_API ZEND_COLD const char *zend_get_object_type(const zend_class_entry *ce)
 		return "trait";
 	} else if (ce->ce_flags & ZEND_ACC_INTERFACE) {
 		return "interface";
+	} else if (ce->ce_flags & ZEND_ACC_ENUM) {
+		return "enum";
 	} else {
 		return "class";
 	}
 }
 /* }}} */
 
+ZEND_API ZEND_COLD const char *zend_get_object_type_uc(const zend_class_entry *ce) /* {{{ */
```

Comment:
I think it's better to have a single parameterized function (it's not that unlikely that other kinds of object types will be introduced in the future), and two respective macros. Maybe:
````.c
ZEND_API ZEND_COLD const char *zend_get_object_type_case(const zend_class_entry *ce, zend_bool uc);
#define zend_get_object_type(ce) zend_get_object_type_case(ce, false);
#define zend_get_object_type_uc(ce) zend_get_object_type_case(ce, true);
````

Reply:
> Do we care about ABI breaks in minor versions?

[ABI breaks are explicitly allowed for minor versions](https://wiki.php.net/rfc/releaseprocess?s[]=release&s[]=process#releases_cycle). This is also catered to by `PHP_API_VERSION` which prohibits loading of extensions which have been compiled for another minor version.

> Also, any rules on when do we do macros over inlined functions?

Well, I don't think there are clear rules. Inlined functions might be preferable nowadays (where they're possible).